engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -263,22 +276,60 @@ def has_no_regcode?(auth_header)
 
         protected
 
+        # rubocop:disable Metrics/PerceivedComplexity
+        # rubocop:disable Metrics/CyclomaticComplexity
+        # rubocop:disable Metrics/AbcSize
         def scc_activate_product
           logger.info "Activating product #{@product.product_string} to SCC"
-          auth = request.headers['HTTP_AUTHORIZATION']
+          auth = nil
+          auth = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
+          mode = nil
           if @system.byos?
-            response = SccProxy.scc_activate_product(@product, auth, params[:token], params[:email])
+            mode = 'byos'
+          elsif !@product.free? && @product.extension?
+            regcode_error_message = 'A registration code is required to activate this product'
+            raise ActionController::TranslatedError.new(regcode_error_message) if (@system.payg? || @system.hybrid?) && params[:token].nil?
+
+            # system is not BYOS and
+            # product is a non free extension
+            # and a registration code is provided
+            mode = 'hybrid'
+            # the extensions must be the same version and arch
```

Comment:
Product extensions not necessarily have the same version as the base.

Reply:
Humm, OK, so this `PAYG + registration code that is not expired + base product that matches the version for the LTSS subscription` from the dev plan, does not apply when activating products ?
My understanding was that it does apply

Reply:
For LTSS this works, but in general extensions can have different versions. We can still change this when more extensions will be supported.